src/main/java/fact/utils/SamplePedestalEvent.java

```diff
@@ -0,0 +1,350 @@
+     *
+     * @param index
+     * @return
+     * @throws IOException
+     */
+    private Data getNoiseEvent(int index) throws IOException {
+        Data dbItem = this.database.get(index);
+        int night = (int)dbItem.get("NIGHT");
+        int runid = (int)dbItem.get("RUNID");
+        int noiseNr = (int)dbItem.get("eventNr");
+        int drs0 = (int)dbItem.get("drs0");
+        int drs1 = (int)dbItem.get("drs1");
+
+        //get the path to the file containing the event
+        String fullpath = this.dataFolder+"/"+createPathFromNightAndRunId(night, runid);
+        fullpath = getPathWithExt(fullpath);
+
+        //get the path to the drs file containing the drs information
+        String drsPath = getDrsPath(this.dataFolder, night, drs0, drs1);
+
+        Data item = null;
+        PreviousEventInfoContainer previousEventInfo = new PreviousEventInfoContainer();
+        try {
+            FITSStream fits = new FITSStream(new SourceURL("file:"+fullpath));
+            fits.init();
+            //calculate how many previous events should be read
+            int prevEventsCount = 20;
+            int noiseNrStartZero = noiseNr - 1; // the noise number starts with 1 but we need start with 0
+            int startPrevEvents = noiseNrStartZero - prevEventsCount;
+            if (startPrevEvents<0) {
+                startPrevEvents = 0;
+                prevEventsCount = noiseNrStartZero;
+            }
+            fits.skipRows(startPrevEvents+1); // add one so the last event is our pedestal
+            for (int i=0; i<prevEventsCount; i++) {
+                item = fits.readNext();
+
+                Utils.isKeyValid(item, "StartCellData", short[].class);
+                Utils.isKeyValid(item, "NROI", Integer.class);
+                Utils.isKeyValid(item, "UnixTimeUTC", int[].class);
+
+                int[] eventTime = (int[]) item.get("UnixTimeUTC");
+                short[] startCellArray = (short[])item.get("StartCellData");
+                int length = (Integer) item.get("NROI");
+
+                short[] stopCellArray = new short[startCellArray.length];
+                //calculate the stopcellArray for the current event
+                for (int j = 0; j < startCellArray.length; ++j) {
+                    //there are 1024 capacitors in the ringbuffer
+                    stopCellArray[j] = (short) ((startCellArray[j] + length) % 1024);
+                }
+
+                previousEventInfo.addNewInfo(startCellArray, stopCellArray, eventTime);
+            }
+        } catch (Exception e) {
+            log.error("Couldn't read file: "+fullpath);
+            throw new RuntimeException(e);
+        }
+
+        if (item==null) {
+            log.error("Couldn't read the pedestal item, nr: '"+noiseNr+"' file: "+fullpath);
+            throw new IOException("Couldn't read the pedestal item");
+        }
+        item.put("noiseNr", noiseNr);
```

Comment:
I'd like to have a more descriptive name instead of noiseNr. I guess it is the event ID of the sampled pedestal event. 
So how about pedestal_event_id, or background_event_id or so